### PR TITLE
Add optional logging support into AP_Periph

### PR DESCRIPTION
**Pull request by peterbarker**
This is 8 bytes larger than master - we now implement the constructor.  I could remove that based on `HAL_LOGGING_ENABLED` too - but I'm pretty sure I've exceeded my `#if` quota already on this PR.

### CODE REVIEW

File: Tools/ardupilotwaf/boards.py [hunk 1]
```diff
@@ -549,6 +549,10 @@ def configure_env(self, cfg, env):
             APJ_BOARD_ID = 100,
             HAL_NO_GCS = 1,
             HAL_LOGGING_ENABLED = 0,
```
Comment (bugobliterator) [inline Tools/ardupilotwaf/boards.py:551]: we should enable logging for sitl_gps I think.

**Review by bugobliterator — CHANGES_REQUESTED** [review]
Looks good to me. Can we enable Logging for sitl_gps_periph?

File: Tools/ardupilotwaf/boards.py (hunk 1 above)
Comment (peterbarker) [reply Tools/ardupilotwaf/boards.py:551]: I didn't want to fiddle with your working setup :-)

One issue I can see with enabling it is that both the main firmware you'll probably be running and the SITL GPS want to log into the same directory!

I think we should consider that problem at a future date?

Comment (bugobliterator) [reply Tools/ardupilotwaf/boards.py:551]: The reason of adding is that it automatically will put your logging changes as part of CI run. But yeah, I don't think this should hold this PR.

**Comment by peterbarker** [comment]
I've split off the code which was seeking to set a RTC timestamp on the created log files; it will be back as part of another PR.

**Comment by peterbarker** [comment]
I've tested this again both in SITL and on the PixRacer-periph firmware and it works as far as logging GPS messages.

**Review by bugobliterator — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Pull request merged**